File: src/main/kotlin/ch/lalumash/IBNG/controllers/JwtAuthenticationController.java
package ch.lalumash.IBNG.controllers;

import ch.lalumash.IBNG.dtos.AuthResponse;
import ch.lalumash.IBNG.entities.JwtRequest;
import ch.lalumash.IBNG.entities.JwtResponse;
import ch.lalumash.IBNG.entities.UserEntity;
import ch.lalumash.IBNG.helper.JwtTokenUtil;
import ch.lalumash.IBNG.services.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.security.authentication.DisabledException;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;
@RestController
@CrossOrigin
public class JwtAuthenticationController {
    @Autowired
    private AuthenticationManager authenticationManager;
    @Autowired
    private JwtTokenUtil jwtTokenUtil;
    @Autowired
    private UserService userService;
    @RequestMapping(value = "/authenticate", method = RequestMethod.POST)
    public AuthResponse createAuthenticationToken(@RequestBody JwtRequest authenticationRequest) throws Exception {
        authenticate(authenticationRequest.getUsername(), authenticationRequest.getPassword());

        final UserDetails userDetails = userService
                .loadUserByUsername(authenticationRequest.getUsername());
        final String token = jwtTokenUtil.generateToken(userDetails);

        UserEntity userById = userService.getUserById(authenticationRequest.getUsername());
        assert userById != null;
        return new AuthResponse(token, userById.getUsername(), userById.getNickname());

    }
    private void authenticate(String username, String password) throws Exception {
        try {
            authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(username, password));
        } catch (DisabledException e) {
            throw new Exception("USER_DISABLED", e);
        } catch (BadCredentialsException e) {
            throw new Exception("INVALID_CREDENTIALS", e);
        }
    }
}

File: src/main/kotlin/ch/lalumash/IBNG/controllers/JwtAuthenticationController.kt
package ch.lalumash.IBNG.controllers

import ch.lalumash.IBNG.dtos.AuthResponse
import ch.lalumash.IBNG.entities.JwtRequest
import ch.lalumash.IBNG.helper.JwtTokenUtil
import ch.lalumash.IBNG.services.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.BadCredentialsException
import org.springframework.security.authentication.DisabledException
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.web.bind.annotation.*

@RestController
@CrossOrigin
class JwtAuthenticationController(
        val authenticationManager: AuthenticationManager,
        val jwtTokenUtil: JwtTokenUtil,
        val userService: UserService
) {
    @RequestMapping(value = ["/authenticate"], method = [RequestMethod.POST])
    @Throws(Exception::class)
    fun createAuthenticationToken(@RequestBody authenticationRequest: JwtRequest): AuthResponse {
        authenticate(authenticationRequest.username!!, authenticationRequest.password!!)
        val userDetails = userService
                ?.loadUserByUsername(authenticationRequest.username)
        val token = jwtTokenUtil!!.generateToken(userDetails)
        val userById = userService!!.getUserById(authenticationRequest.username!!)!!
        return AuthResponse(token, userById.username, userById.nickname)
    }

    @Throws(Exception::class)
    private fun authenticate(username: String, password: String) {
        try {
            authenticationManager!!.authenticate(UsernamePasswordAuthenticationToken(username, password))
        } catch (e: DisabledException) {
            throw Exception("USER_DISABLED", e)
        } catch (e: BadCredentialsException) {
            throw Exception("INVALID_CREDENTIALS", e)
        }
    }
}